hack/make-rules/clean.sh

```diff
@@ -29,7 +29,7 @@ CLEAN_PATTERNS=(
 )
 
 for pattern in "${CLEAN_PATTERNS[@]}"; do
-  while IFS=$'\n' read -r -d match; do
+  while IFS=$'\n' read -r match; do
     echo "Removing ${match#${KUBE_ROOT}\/} .."
     rm -rf "${match#${KUBE_ROOT}\/}"
   done <   <(find "${KUBE_ROOT}" -iregex "^${KUBE_ROOT}/${pattern}$")
```

Comment:
We should probably use `-print0` and use this as the delimiter.

Reply:
@BenTheElder do you mean (1) eliminate `IFS` and (2) append `-print0` in "find" as below?

```diff
diff --git a/hack/make-rules/clean.sh b/hack/make-rules/clean.sh
index 9305c8df6c..9e85c3f879 100755
--- a/hack/make-rules/clean.sh
+++ b/hack/make-rules/clean.sh
@@ -29,10 +29,10 @@ CLEAN_PATTERNS=(
 )
 
 for pattern in "${CLEAN_PATTERNS[@]}"; do
-  while IFS=$'\n' read -r match; do
+  while read -r match; do
     echo "Removing ${match#${KUBE_ROOT}\/} .."
     rm -rf "${match#${KUBE_ROOT}\/}"
-  done <   <(find "${KUBE_ROOT}" -iregex "^${KUBE_ROOT}/${pattern}$")
+  done < <(find "${KUBE_ROOT}" -iregex "^${KUBE_ROOT}/${pattern}$" -print0)
 done
 
 # ex: ts=2 sw=2 et filetype=sh

```